From ea8082b75f5a0938f8f3d50e683db6870c213e1b Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Thu, 10 Mar 2022 05:24:03 +0200
Subject: [PATCH] [SW-77829] rom: update image descriptor

Due to the new design, the sha for ppboot and preboot images need to
be stored in the image descriptor for image validation.

Due to this the checksum copy for SPI images needs to be enabled in
boot desc read api.

Change-Id: I4cf9fd300ab3490fe409ab26fc415e93c772fda3
---

diff --git a/common/boot_desc.c b/common/boot_desc.c
index 1cf719a..2d7450e 100644
--- a/common/boot_desc.c
+++ b/common/boot_desc.c
@@ -151,9 +151,8 @@
 			p_img_desc[i].start_high[idx] << 32) |
 			p_img_desc[i].start_low[idx];
 
-		if (img != HL_FW_BOOT_PTR_SPI)
-			memcpy(desc->chksum, p_img_desc[i].chksum2[idx],
-				   ARRAY_SIZE(desc->chksum));
+		memcpy(desc->chksum, p_img_desc[i].chksum2[idx],
+			   ARRAY_SIZE(desc->chksum));
 
 		return idx;
 	}
